### PR TITLE
core: riscv: Support fault mitigations in non-threaded code

### PR DESCRIPTION
**Pull request by gagachang**
The previous commit
ce56605a0ede ("core: support fault mitigations in non-threaded code") supports fault mitigations in non-threaded code for ARM architecture.

This commit adds the related modification for RISC-V architecture to support it too.

### CODE REVIEW

**Comment by gagachang** [comment]
Hi @maroueneboubakri 

This PR fixes compiler error for upstream RISC-V.

File: lib/libutils/ext/include/fault_mitigation.h [hunk 1]
```diff
@@ -250,7 +250,7 @@ extern struct ftmn_func_arg *__ftmn_global_func_arg;
 static inline struct ftmn_func_arg **__ftmn_get_tsd_func_arg_pp(void)
 {
 #if defined(CFG_FAULT_MITIGATION) && defined(__KERNEL__)
-    if (thread_get_id_may_fail() >= 0)
+	if (thread_get_id_may_fail() >= 0)
```
Comment (jenswi-linaro) [inline lib/libutils/ext/include/fault_mitigation.h:253]: I'd prefer to see this in a separate commit since it's a generic change, not a riscv change.

Comment (gagachang) [reply lib/libutils/ext/include/fault_mitigation.h]: I submitted another PR https://github.com/OP-TEE/optee_os/pull/6356

**Comment by gagachang** [comment]
Hello @jenswi-linaro @jforissier 
Could we merge this minor PR?

**Comment by jforissier** [comment]
`Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Comment by jforissier** [comment]
> Hello @jforissier I can restart IBART but it always fails with mbedtls related source files. Would you have time to take a look at it?

You need to rebase your branch onto master. The failure is because IBART does not rebase or merge automatically, whereas GitHub actions do.

**Comment by gagachang** [comment]
> You need to rebase your branch onto master. The failure is because IBART does not rebase or merge automatically, whereas GitHub actions do.

Thank you! Now IBART passes.

**Pull request merged**